src/map/script.cpp

```diff
@@ -9339,9 +9287,7 @@ BUILDIN_FUNC(autobonus3)
 
 	rate = script_getnum(st,3);
 	dur = script_getnum(st,4);
-	data = script_getdata(st, 5);
-	get_val(st, data); // Convert into value in case of a variable
-	atk_type = ( data_isstring(data) ? skill_name2id(script_getstr(st,5)) : script_getnum(st,5) );
+	atk_type = ( script_isstring(st, 5) ? skill_name2id(script_getstr(st,5)) : script_getnum(st,5) );
```

Comment:
Skill ID is not validated. skill_name2id returns 0 in case it does not exist.
Bad variable name as well.